PGP/INLINE and build in combined mime format

Valodim opened the pull request:
This PR supersedes both #1198 and #1323.

It adds PGP/INLINE, where the UI has been unchanged since #1198. It also changes PgpMessageBuilder to use combined mime format, adding tests and solving some bugs.

philipwhiuk commented on k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java:
```diff
@@ -260,6 +257,11 @@ public void onCryptoModeChanged(CryptoMode cryptoMode) {
+        recipientPresenter.onCryptoPgpInlineChanged(enabled);
+    }
+
     enum Action {
```
I'm a bit sad about the reference to specific crypto methods in MessageCompose.

Valodim replied:
Correct, but there isn't really a much nicer way for getting a callback from the PgpInlineDialog through to the presenter other than having the activity implement an interface. It's also consistent with the way `onCryptoModeChanged` is handled.

philipwhiuk commented on k9mail/src/main/res/layout/openpgp_inline_dialog.xml:
```diff
@@ -0,0 +1,122 @@
+            android:src="@drawable/compatibility"
+            android:scaleType="fitCenter"
+            android:tint="@color/light_black"
+            />
```
It would be nice if we could move towards setting `android:contentDescription` (http://developer.android.com/training/accessibility/accessible-app.html) or `tools:ignore="contentDescription"` for ImageView elements.

Personally I'd like us to move towards builds failing on lint warnings. But in the meantime it would be nice if we didn't make more of them :)

I don't mind it going in without this given there's so many already though. We can swing through the whole lot adding them.

Valodim replied:
little point thinking about this when we plan to do a design overhaul sooner or later

philipwhiuk commented on k9mail/src/main/res/values/colors.xml:
```diff
@@ -1,7 +1,11 @@
     <color name="message_list_item_footer_background">#eeeeee</color>
-    
+
+    <color name="light_black">#444</color>
```
light black, green and red don't really help knowing where to use it. 

Have you tested how the 'light black' looks on Dark Theme?

philipwhiuk commented on k9mail/src/main/res/values/styles.xml:
```diff
@@ -51,5 +51,9 @@
 
+    <style name="ShowcaseTheme" parent="ShowcaseView.Light">
+        <item name="sv_backgroundColor">#99444444</item>
+    </style>
```
Do we need ShowcaseTheme.Light and ShowcaseTheme.Dark.

cketti commented:
In the future let's try to name colors/drawables after function, not appearance.
Examples: `circled_minus` could become `negative_bullet_point`. Instead of `light_red` maybe use `negative_bullet_point_tint` and make it a reference to `light_red`.

`AssertionError` messages don't need to contain the text "this is a bug". It's implied by the type.

Minor style issue: Error/exception messages should be properly capitalized.

I'm happy to merge if you're fine with my changes. We can fix the theme issues later.

Valodim commented:
changes look good to me.

I removed some "this is a bug" mentions, changed the drawable name (as a fixup, to avoid crummy history),  moved the bullet point appearance into a style, and the tintColor into the themes (although it's the same color in both right now).
